lib/transaction.js

```diff
@@ -99,8 +102,24 @@ class Transaction {
   }
 
   prepareEnvironment() {
+    let connectionPromise;
 
-    return Utils.Promise.resolve(this.parent ? this.parent.connection : this.sequelize.connectionManager.getConnection({ uuid: this.id }))
+    if (this.parent) {
```

Comment:
I feel this can be improved a whole lot

```js
const acquireOptions = { uuid: this.id };

if (this.options.readOnly) {
  acquireOptions.type = 'SELECT';
}

return Utils.Promise.resolve(
  this.parent.connection
    ? this.parent.connection
    : this.sequelize.connectionManager.getConnection(acquireOptions)
)
```